Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -1703,12 +1703,35 @@ void WebProcess::nonVisibleProcessEarlyMemoryCleanupTimerFired()
     if (!m_pagesInWindows.isEmpty())
         return;
 
+    destroyDecodedDataForAllImages();
+
 #if PLATFORM(COCOA)
     destroyRenderingResources();
     releaseSystemMallocMemory();
 #endif
 }
 
+void WebProcess::destroyDecodedDataForAllImages()
+{
+#if PLATFORM(COCOA)
+    for (auto& page : m_pageMap.values()) {
+        if (!page->isUsingUISideCompositing())
+            return;
+    }
```

Comment:
I do not understand this code. `DrawingArea::setNextRenderingUpdateRequiresSynchronousImageDecoding()` is virtual and it is only implemented in RemoteLayerTreeDrawingArea. So the for-loop below is doing nothing unless the page has a `DrawingArea` of type `RemoteLayerTreeDrawingArea`. Also the mix of having `PLATFORM(COCOA)` and  `isUsingUISideCompositing()` in this function and having `ENABLE(GPU_PROCESS)` in `WebPage::willDestroyDecodedDataForAllImages()` makes it difficult to see when this code can be enabled.

Reply:
I think we want to destroy the decoded image data only when UI side compositing is being used. Otherwise, because `setNextRenderingUpdateRequiresSynchronousImageDecoding` does nothing on a `TiledCoreAnimationDrawingArea`, we will see a flash of unrendered images. Once we implement `setNextRenderingUpdateRequiresSynchronousImageDecoding` on `TiledCoreAnimationDrawingArea` we can remove this check.

An alternative would be to make `willDestroyDecodedDataForAllImages` return whether the decoded data destruction is allowed.

The `PLATFORM(COCOA)` is indeed not needed. (In a previous iteration of the patch I didn't upload, I was checking the drawing area type, and `DrawingAreaType::RemoteLayerTree` is only defined on `PLATFORM(COCOA)`.)